### PR TITLE
Add inputs & buttons from blade-components

### PR DESCRIPTION
**Pull request by Roene-JustBetter**
- Added inputs and buttons from the [Rapidez/blade-components ](https://github.com/rapidez/blade-components).
- Remove old buttons and inputs
- Fixed z-index for popup. 

Examples: 
Scherm­afbeelding 2024-12-03 om 13 47 14
Scherm­afbeelding 2024-12-03 om 13 47 21

### CODE REVIEW

**Comment by Jade-GG** [comment]
Do we still need the actual buttons in here if they're already in `rapidez/blade-components`?

**Comment by Roene-JustBetter** [comment]
> Do we still need the actual buttons in here if they're already in `rapidez/blade-components`?

Yes the `button/cart` we use for all the add to cart buttons. For the base we added the `disableWhenLoading`

File: resources/views/cart/coupon/add.blade.php [hunk 1]
```diff
@@ -10,7 +10,6 @@
 >
     <form v-on:submit.prevent="mutate" class="flex gap-3">
         <x-rapidez::input
-            :label="false"
```
Comment (royduin) [inline resources/views/cart/coupon/add.blade.php:13]: There is also a `wrapperClass` which doesn't do anything anymore.

File: resources/js/stores/useCart.js [hunk 2]
```diff
@@ -271,9 +271,9 @@ export const taxTotal = computed(() => {
...
 })
 
-watch(mask, refresh)
-if (cartStorage.value?.id && !mask.value) {
-    clear()
-}
+// watch(mask, refresh)
+// if (cartStorage.value?.id && !mask.value) {
+//     clear()
+// }
```
Comment (royduin) [inline resources/js/stores/useCart.js:277]: Why? 😇

File: resources/views/cart/item/quantity.blade.php [hunk 3]
```diff
@@ -9,7 +9,6 @@
         <x-rapidez::input
             name="qty"
             type="number"
-            :label="false"
             v-model="variables.quantity"
             v-bind:dusk="'qty-'+index"
             wrapperClass="flex flex-1"
```
Comment (royduin) [inline resources/views/cart/item/quantity.blade.php:14]: `wrapperClass` 😇

File: resources/views/checkout/steps/credentials.blade.php
Comment (royduin) [inline resources/views/checkout/steps/credentials.blade.php:1]: It's back? Previously it was removed in: https://github.com/rapidez/core/pull/657

File: resources/views/checkout/steps/payment.blade.php
Comment (royduin) [inline resources/views/checkout/steps/payment.blade.php:1]: Same, do we need it or not? https://github.com/rapidez/core/pull/657

File: resources/views/product/partials/quantity.blade.php [hunk 4]
```diff
@@ -1,12 +1,10 @@
-<x-rapidez::select
+<x-rapidez::input.select
     name="qty"
     label="Quantity"
     v-model="addToCart.qty"
-    class="w-auto"
-    labelClass="flex items-center sr-only mr-3"
-    wrapperClass="flex"
```
Comment (royduin) [inline resources/views/product/partials/quantity.blade.php:7]: Maybe a search through the project to check if we don't have those anymore?

**Review by royduin — CHANGES_REQUESTED** [review]

File: resources/js/stores/useCart.js (hunk 2 above)
Comment (Roene-JustBetter) [reply resources/js/stores/useCart.js]: Whoops :)

File: resources/views/checkout/steps/payment.blade.php
Comment (Roene-JustBetter) [reply resources/views/checkout/steps/payment.blade.php:1]: I think something went wrong with a merge here. This file is not needed I removed it.

File: resources/views/product/partials/quantity.blade.php (hunk 4 above)
Comment (Roene-JustBetter) [reply resources/views/product/partials/quantity.blade.php:7]: Done nothing found anymore after removing this.

File: config/rapidez/frontend.php [hunk 5]
```diff
@@ -82,6 +82,7 @@
     // Z-Index per element
     'z-indexes' => [
         'header-dropdowns' => 'z-20',
+        'popup'            => 'z-20',
```
Comment (JimmyHoenderdaal) [inline config/rapidez/frontend.php:85]: Do we need this? We already have lightbox

File: resources/views/cart/item/quantity.blade.php [hunk 6]
```diff
@@ -9,19 +9,16 @@
         <x-rapidez::input
             name="qty"
             type="number"
-            :label="false"
             v-model="variables.quantity"
             v-bind:dusk="'qty-'+index"
-            wrapperClass="flex flex-1"
-            {{-- TODO: We don't need these importants with Tailwind merge and center isn't really center with type number --}}
-            class="!w-14 !px-1 text-center"
+            class="w-14 !px-1 text-center"
```
Comment (JimmyHoenderdaal) [inline resources/views/cart/item/quantity.blade.php:14]: Previous todo said we didn't need importants, do you really need this one?

File: resources/views/layouts/partials/footer/newsletter.blade.php [hunk 7]
```diff
@@ -26,7 +23,7 @@ class="size-full min-w-0 appearance-none rounded-md border bg-white py-2 px-4 te
                                 <x-rapidez::button.secondary
                                     type="submit"
                                     dusk="newsletter-submit"
-                                    class="w-full"
+                                    class="w-full static"
```
Comment (JimmyHoenderdaal) [inline resources/views/layouts/partials/footer/newsletter.blade.php:26]: What do you need static for?

**Review by JimmyHoenderdaal — CHANGES_REQUESTED** [review]

File: resources/views/layouts/partials/footer/newsletter.blade.php (hunk 7 above)
Comment (Roene-JustBetter) [reply resources/views/layouts/partials/footer/newsletter.blade.php]: Not needed removed it.

**Review by jordythevulder — APPROVED** [review]

File: resources/views/product/partials/gallery/popup.blade.php [hunk 8]
```diff
@@ -1,4 +1,4 @@
-<div v-if="images.length && zoomed" class="fixed inset-0 bg-white cursor-zoom-out z-popup flex">
+<div v-if="images.length && zoomed" class="fixed inset-0 bg-white cursor-zoom-out flex {{ config('rapidez.frontend.z-indexes.popup') }}">
```
Comment (royduin) [inline resources/views/product/partials/gallery/popup.blade.php:1]:
```suggestion
<div v-if="images.length && zoomed" class="fixed inset-0 bg-white cursor-zoom-out flex z-popup">
```

**Pull request merged**